src/sync/incomingSyncUtils.js

```diff
@@ -277,6 +277,14 @@ export const sanityCheckIncomingRecord = (recordType, record) => {
       ],
       canBeBlank: [],
     },
+    NameTag: {
+      cannotBeBlank: ['ID'],
+      canBeBlank: ['description'],
+    },
+    NameTagJoin: {
+      cannotBeBlank: ['ID', 'name_ID', 'name_tag_ID'],
+      canBeBlank: [],
+    },
```

Comment:
This mechanism is ancient, 2016 times with me and Edwin. I can't remember what the point of `canBeBlank` is lol. Like it either can or can't, so the `cannotBeBlank` kind of covers it. There are heaps of fields not specified for so many table haha.

Like I wonder if we could have ignored this and used realms mechanism (optional field) and errors for required fields?

Not actionable here, but worth making an issue ya reckon? I'd be stoked if we could scrap 300 lines or so of code.

Reply:
I think refactoring sync would be a top 3 on my todo list of biggest impact tasks for mobile! 
Removing this would be nice and an issue would be good, but lower priority than say, factoring the sync path to be more testable